src/Couchbase.Lite.Tests.Shared/ReplicationTest.cs

```diff
@@ -1291,6 +1319,107 @@ public void TestConflictResolverExceptionsNoneMatchDBThrown()
+
+            using (var doc = Db.GetDocument("doc1")) {
+                doc.GetBlob("blob")?.Content.Should().ContainInOrder(new byte[] { 6, 6, 6 });
+            }
+        }
+
+        [Fact]
+        public void TestConflictResolverReturningBlobFromDifferentDB()
+        {
+            var tmpDoc = new MutableDocument("doc1");
+            var evilByteArray = new byte[] { 6, 6, 6 };
+            using (var thirdDb = new Database("different_db")) {
+                tmpDoc.SetBlob("blob", new Blob("text/plaintext", evilByteArray));
+                thirdDb.Save(tmpDoc);
+
+                var differentDbResolver = new TestConflictResolver((conflict) => tmpDoc);
+
+                TestConflictResolverExceptionThrown(differentDbResolver, false);
+
+                thirdDb.Delete();
+            }
+        }
+
+        private void TestConflictResolverWinsInBlobs(bool returnRemoteDoc)
+        {
+            CreateReplicationConflictInBlobs();
+
+            var config = CreateConfig(false, true, false);
+
+            config.ConflictResolver = new TestConflictResolver((conflict) =>
+            {
+                if (returnRemoteDoc) {
+                    return conflict.RemoteDocument;
+                } else
+                    return conflict.LocalDocument;
+            });
+
+            RunReplication(config, 0, 0);
+
+            using (var doc = Db.GetDocument("doc1")) {
+                if (returnRemoteDoc) {
+                    doc.GetBlob("blob")?.Content.Should().ContainInOrder(new byte[] { 7, 7, 7 });
+                } else {
+                    doc.GetBlob("blob")?.Content.Should().ContainInOrder(new byte[] { 4, 5, 6 });
+                }
+            }
+        }
+
+        private void CreateReplicationConflictInBlobs()
+        {
+            var oddByteArray = new byte[] { 1, 2, 3 };
+            using (var doc1 = new MutableDocument("doc1")) {
+                doc1.SetBlob("blob", new Blob("text/plaintext", oddByteArray));
+                Db.Save(doc1);
+            }
+
+            using (var doc1 = new MutableDocument("doc1")) {
+                doc1.SetBlob("blob", new Blob("text/plaintext", oddByteArray));
+                _otherDB.Save(doc1);
+            }
+
+            // Force a conflict
+            using (var doc1a = Db.GetDocument("doc1"))
+            using (var doc1aMutable = doc1a.ToMutable()) {
+                var evenByteArray = new byte[] { 4, 5, 6 };
+                doc1aMutable.SetBlob("blob", new Blob("text/plaintext", evenByteArray));
+                Db.Save(doc1aMutable);
+            }
+
+            Db.Count.ShouldBeEquivalentTo(1);
+
+            using (var doc1a = _otherDB.GetDocument("doc1"))
+            using (var doc1aMutable = doc1a.ToMutable()) {
+                var luckyByteArray = new byte[] { 7, 7, 7 };
+                doc1aMutable.SetBlob("blob", new Blob("text/plaintext", luckyByteArray));
+                _otherDB.Save(doc1aMutable);
+            }
+        }
+
```

Comment:
This seems like a lot of duplication.  Is there a way to refactor the original method to optionally use blobs?